functions/http/src/main/java/com/example/functions/http/ParseContentType.java

```diff
@@ -26,44 +26,52 @@
 import java.io.BufferedWriter;
 import java.io.IOException;
 import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
 import java.util.Base64;
+import java.util.Optional;
 
 public class ParseContentType implements HttpFunction {
 
   // Use GSON (https://github.com/google/gson) to parse JSON content.
-  private Gson gsonParser = new Gson();
+  private static final Gson gson = new Gson();
 
   // Responds to an HTTP request using data from the request body parsed according to the
   // "content-type" header.
   @Override
   public void service(HttpRequest request, HttpResponse response)
       throws IOException {
-    String name = null;
+    String name;
     String contentType = request.getContentType().get();
```

Comment:
maybe track for a future issue. getContentType returns an optional, so contentType may not be present.

```java
request.getContentType().ifPresent(contentType -> {
});
```

or
```java
String contentType = request.getContentType().orElseThrow(() -> new RuntimeException("content type not specified"));
```

Reply:
Good catch - will refactor.

(Aside: it looks like `Content-Type` values are [_recommended_ but not _required_](https://stackoverflow.com/questions/20392345/is-content-type-http-header-always-required), at least in **reponses**.)

Reply:
FYI: specified a _default value_ instead, and included a comment as to why. (`null` breaks the `switch...case`, and `get` on a missing `Optional` [throws a `NoSuchElementException`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--))

```java
// Default values avoid null issues (with switch/case) and exceptions from get() (optionals)
String contentType = request.getContentType().orElse("");
```

LMK if this is too much "black magic" to include in a sample.